File: api/pedidos.js
export default async function handler(req, res) {
  // --- Configuração CORS ---
  // Permite requisições do seu domínio do GitHub Pages
  res.setHeader('Access-Control-Allow-Origin', 'https://nicholas-apolinario03.github.io);
  // Permite os métodos HTTP que você pode usar (GET, POST, OPTIONS, etc.)
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS, PUT, PATCH, DELETE');
  // Permite os cabeçalhos que o navegador pode enviar
  res.setHeader('Access-Control-Allow-Headers', 'X-Requested-With,content-type,Authorization');

  // Lida com requisições OPTIONS (preflight requests) que os navegadores fazem para CORS
  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }
  // --- Fim da Configuração CORS ---

  const { order_id } = req.query;

  if (!order_id) {
    return res.status(400).json({ error: "Parâmetro 'order_id' obrigatório." });
  }

  const access_token = process.env.MELI_ACCESS_TOKEN;

  if (!access_token) {
    return res.status(500).json({ error: "Access token não configurado." });
  }

  const url = `https://api.mercadolibre.com/orders/${order_id}`;

  try {
    const response = await fetch(url, {
      headers: {
        Authorization: `Bearer ${access_token}`,
        'x-version': '2'
      }
    });

    const data = await response.json();

    if (response.ok) {
      return res.status(200).json(data);
    } else {
      return res.status(response.status).json({
        error: "Erro ao buscar pedido.",
        message: data
      });
    }
  } catch (error) {
    return res.status(500).json({
      error: "Erro interno.",
      message: error.message
    });
  }
}

File: api/pedidos.py
from http.server import BaseHTTPRequestHandler
import json
import requests

# Substitua por seu token de acesso
ACCESS_TOKEN = 'APP_USR-3704242181199025-052809-ef0f7bd5efacd9e0df113aa937612b1c-1357030258'

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        # Pegando parâmetros da URL, ex.: ?order_id=123456
        from urllib.parse import parse_qs, urlparse
        query_components = parse_qs(urlparse(self.path).query)
        order_id = query_components.get("order_id", [None])[0]

        if order_id is None:
            self.send_response(400)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({"error": "Parâmetro 'order_id' obrigatório."}).encode())
            return

        url = f"https://api.mercadolibre.com/orders/{order_id}"
        headers = {
            'Authorization': f'Bearer {ACCESS_TOKEN}',
            'x-version': '2'
        }

        response = requests.get(url, headers=headers)

        if response.status_code == 200:
            pedido = response.json()
            self.send_response(200)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps(pedido, ensure_ascii=False).encode())
        else:
            self.send_response(response.status_code)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({
                "error": f"Erro ao buscar pedido. Status {response.status_code}",
                "message": response.text
            }).encode())
